atom/common/native_mate_converters/net_converter.cc

```diff
@@ -191,16 +191,29 @@ bool Converter<net::HttpResponseHeaders*>::FromV8(
     if (!keys->Get(i)->ToString(context).ToLocal(&key)) {
       return false;
     }
-    if (!headers->Get(key)->ToString(context).ToLocal(&value)) {
-      return false;
+
+    if (headers->Get(key)->IsArray()) {
+      auto values = v8::Local<v8::Array>::Cast(headers->Get(key));
+      for (uint32_t j = 0; j < values->Length(); j++) {
+        if (!values->Get(j)->ToString(context).ToLocal(&value)) {
+          return false;
+        }
+
+        std::string k, v;
+        mate::ConvertFromV8(isolate, key, &k);
+        mate::ConvertFromV8(isolate, value, &v);
+        out->AddHeader(k + ": " + v);
+      }
+    } else {
+      if (!headers->Get(key)->ToString(context).ToLocal(&value)) {
+        return false;
+      }
+
+      std::string k, v;
+      mate::ConvertFromV8(isolate, key, &k);
```

Comment:
_(minor)_ we're converting `key` on each array element, when we only ever need to convert it once.